packages/memory-storage/test/write-metadata.test.ts

```diff
@@ -20,11 +20,9 @@ describe('writeMetadata option', () => {
         test('creating a data store should not write __metadata__.json file', async () => {
             const keyValueStore = await storage.keyValueStores().getOrCreate();
             const expectedPath = resolve(storage.keyValueStoresDirectory, `${keyValueStore.id}`);
-            await waitTillWrittenToDisk(expectedPath);
-
-            const directoryFiles = await readdir(expectedPath);
 
-            expect(directoryFiles).toHaveLength(0);
+            // We check that reading the directory for the store throws an error, which means it wasn't created on disk
+            await expect(() => readdir(expectedPath)).rejects.toThrow();
```

Comment:
FYI the argument does not have to be callback (as long as its async), this should work the same:

```suggestion
            await expect(readdir(expectedPath)).rejects.toThrow();
```

At least that is how I use it, if there is a reason to prefer the callback approach, let me know :]